subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolutionstrategy/DefaultCachePolicy.java

```diff
@@ -149,14 +151,31 @@ private void eachArtifact(Action<? super ArtifactResolutionControl> rule) {
         artifactCacheRules.add(0, rule);
     }
 
+    /**
+     * @param resolutionControl The resolution control to mutate
+     * @return If the offline or refresh dependencies rule was applied to the provided resolution control.
+     */
+    private boolean applyOfflineAndRefreshRules(ResolutionControl<?, ?> resolutionControl) {
```

Comment:
❕ I don't think this is a very cohesive method.  It applies either offline _or_ refresh, never both.  Also, the `true` return doesn't tell you which one it did.  I'd consider splitting this into 2 methods for either case and use trying both using `||` in place of calling this.